software-development/SDLC.md

```diff
@@ -0,0 +1,101 @@
+Testing involves not only testing the new functionality that is build, but also making sure the other parts of a system still work as planned. For example if you need to update the schema of your database while developing a new feature, you want to make sure this doesn’t have any negative consequences for other parts of the system. This type of testing is called regreission testing. 
+From Wikipedia: *Regression testing is re-running functional and non-functional tests to ensure that previously developed and tested software still performs after a change. If not, that would be called a regression.* <br/>
+
+**Who is responsible:** depending on the organization, separate Quality assurance testers, test engineers or the engineers themselves.
+
+
+## Deployment (and environments)
+
+**Background**
+When working in bigger organizations you will notice that there is more than one environment that is being used to develop and deliver software in a controllable manner. Before we explain a bit on the concepts and the different environments let’s do a small step back and align on a deployment.
+
+{% hyf-youtube src="https://www.youtube.com/watch?v=nZ3BaTY8c9M" %}
+
+In the explanation above you see the difference of a local development environment and the environment which runs your software for the rest of the world to see (the production environment). However when multiple people work on the same code base and software program grows in complexity organizations tend to add one or more environments in the process.
+
+
+**Different environments** <br/>
+**Development:**  This is where the software is developed, this could be the laptop of the engineer or a shared server where multiple engineers are working on together.  <br/>
+**Testing:** this is the environment where the main testing is performed. From the 4 environments described this environment is the one that is sometimes optional. Testing could also be performed in both the development and Staging environment. <br/>
+**Staging:** This environment (sometimes called acceptance) is a near to identical version of production (usually without the customer data) to validate if everything is running in the right way. As this environment has the same setup as production things like performance test (determining if the new release will reduce the maximum  throughput) or security testing (to determine if any vulnerabilities are introduced) are usually performed. <br/>
+**Production:** This is the environment that the customers are actually using. <br/>
```

Comment:
Maybe add that this environment is protected and developers should never push to it?